driver-core/src/test/functional/com/mongodb/operation/CreateCollectionOperationSpecification.groovy

```diff
@@ -185,6 +185,29 @@ class CreateCollectionOperationSpecification extends OperationFunctionalSpecific
         }
     }
 
+    def 'should create collection in respect to the autoIndex option'() {
+        given:
+        assert !collectionNameExists(getCollectionName())
+
+        when:
+        new CreateCollectionOperation(getDatabaseName(), getCollectionName())
+                .autoIndex(autoIndex)
+                .execute(getBinding())
+
+        then:
+        def stats = new CommandWriteOperation<Document>(getDatabaseName(),
+                new BsonDocument('collStats', new BsonString(getCollectionName())),
+                new DocumentCodec()).execute(getBinding())
+
+        expect:
```

Comment:
Sorry, I'm not familiar with Spock, it's seems to me that all the test file should be rewritten from what you said because it's using the pattern I've been copying from. This change should be in another pull request since it's not part of the bug fixing.

Thanks,



Thanks,